pkg/kbld/builder/bazel/bazel.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package image
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os/exec"
+	"regexp"
+
+	ctlbdk "github.com/k14s/kbld/pkg/kbld/builder/docker"
+	"github.com/k14s/kbld/pkg/kbld/config"
+	ctllog "github.com/k14s/kbld/pkg/kbld/logger"
+)
+
+var (
+	// Example output that includes the final digest:
+	// INFO: Analyzed target //:simple-app (0 packages loaded, 0 targets configured).
+	// INFO: Found 1 target...
+	// Target //:simple-app up-to-date:
+	//   bazel-bin/simple-app-layer.tar
+	// INFO: Elapsed time: 0.148s, Critical Path: 0.00s
+	// INFO: 1 process: 1 internal.
+	// INFO: Build completed successfully, 1 total action
+	// INFO: Build completed successfully, 1 total action
+	// Loaded image ID: sha256:328b5f47550c85cea5284911ad4d284ce20e8240d61d2610eb6cb4aa8b43c19e
+	// Tagging 328b5f47550c85cea5284911ad4d284ce20e8240d61d2610eb6cb4aa8b43c19e as bazel:simple-app
+	bazelImageID = regexp.MustCompile("Loaded image ID: (sha256:)([0-9a-z]+)")
+)
+
+type Bazel struct {
+	docker ctlbdk.Docker
+	logger ctllog.Logger
+}
+
+func NewBazel(docker ctlbdk.Docker, logger ctllog.Logger) Bazel {
+	return Bazel{docker: docker, logger: logger}
+}
+
+func (b *Bazel) Build(image, directory string, opts config.SourceBazelBuildOpts) (ctlbdk.DockerTmpRef, error) {
+	prefixedLogger := b.logger.NewPrefixedWriter(image + " | ")
+
+	prefixedLogger.Write([]byte(fmt.Sprintf("starting build (using bazel): %s\n", directory)))
+	defer prefixedLogger.Write([]byte("finished build (using bazel)\n"))
+
+	var imageID string
+	{
+		var stdoutBuf, stderrBuf bytes.Buffer
+
+		cmdArgs := []string{"run"}
```

Comment:
Are we deliberately doing a "bazel [run](https://docs.bazel.build/versions/4.2.1/user-manual.html#run)" rather than a "bazel [build](https://docs.bazel.build/versions/4.2.1/guide.html#the-build-command)"?

Reply:
oh... I think I answered my own question.

"bazel build" produces (among other artifacts) a script that will config, (docker) load, and tag the image (e.g. `bazel-bin/simple-app.executable`), but that script is not executed.
"bazel run" executes the script. 👍🏻

Reply:
🙏🏻